kube/src/config/mod.rs

```diff
@@ -91,11 +91,17 @@ impl Config {
     /// and relies on you having the service account's token mounted,
     /// as well as having given the service account rbac access to do what you need.
     pub fn from_cluster_env() -> Result<Self> {
-        let cluster_url = incluster_config::kube_server().ok_or(ConfigError::MissingInClusterVariables {
-            hostenv: incluster_config::SERVICE_HOSTENV,
-            portenv: incluster_config::SERVICE_PORTENV,
-        })?;
-        let cluster_url = cluster_url.parse::<http::Uri>()?;
+
+        let cluster_url = if cfg!(feature = "rustls-tls") {
```

Comment:
We still need to detect whether we're in-cluster somehow, so that we can fall back to kubeconfig as needed.

Reply:
This method is only dealing with incluster auth. The fallback happens above in `Config::infer`.

Reply:
`Config::infer` just looks at whether `from_cluster_env` returned `Ok`.

Reply:
And if it's not an `Ok`, we will fall back to a local kubeconfig. Are you worried that because we are looking at less evars that we might get an unintentional `Ok` and fail to fallback? We still have to find mounted certs and token files.

Reply:
Oh, should be ok then.